RoddyCore/test/de/dkfz/roddy/config/ToolFileParameterTest.groovy

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2017 eilslabs.
+ *
+ * Distributed under the MIT License (license terms are at https://www.github.com/eilslabs/Roddy/LICENSE.txt).
+ */
+
+package de.dkfz.roddy.config
+
+import de.dkfz.roddy.knowledge.files.BaseFile
+import groovy.transform.CompileStatic;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by heinold on 08.03.17.
+ */
+@CompileStatic
+public class ToolFileParameterTest {
+    @Test
+    public void testCloneAndEquals() throws Exception {
+        def src = new ToolFileParameter(BaseFile, [], "ABC", new ToolFileParameterCheckCondition(true), "DEF", null, "abc")
+        def clone = src.clone()
+        assert clone == src
```

Comment:
AFAIKT in Java .equals is **per default** set to ==. But indeed you are right in Groovy the situation is different. But apparently it is not as simple as you said (and is written here by mrhaki in 2009: http://mrhaki.blogspot.de/2009/09/groovy-goodness-check-for-object.html). 

See for instance this somewhat newer post: http://stackoverflow.com/questions/13687154/groovy-operator.

I guess we should have a look at the source code for equals and ==. :-|